pos_customer_required/models/pos_config.py

```diff
@@ -18,8 +18,8 @@ class PosConfig(models.Model):
     ]
 
     require_customer = fields.Selection(
-        selection=_REQUIRE_CUSTOMER_KEYS,
-        string='Require Customer',
+        selection=_REQUIRE_CUSTOMER_KEYS, required=True,
```

Comment:
Don't make it required. It's not needed.

Reply:
Well, it was an error in fact. if it's not required 4 value are possible, and not 3. 
I don't understand what is the problem, if a default value is set. 
thanks for your answer.

Reply:
It may be either required False and two options in Selection, or required True and three options in selection. I'd prefer required=True

Reply:
Adding required fields on existing models makes tests of other modules that inherit from _point_of_sale_ to not work due to a not null constraint in the DB and the code that defaults it to the correct value not yet loaded. It's what it has happened with all the *_warn fields, and Odoo has removed it the `required` flag on them on master. I prefer to avoid this.

Reply:
FYI, I have just found the problem on one of l10n-spain modules and remove it: https://github.com/OCA/l10n-spain/commit/07f182134896976bac42fda92958c889030539bf

Reply:
Please @legalsylvain could you check this last comment?

Reply:
@pedrobaeza is it still true when we have ``default='value'`` attribute, i.e. does odoo adds sql ``DEFAULT`` attribute in case of having static default value?

Reply:
ok, it seems that odoo doesn't use ``DEFAULT`` on db level